src/System.CommandLine/Parsing/StringExtensions.cs

```diff
@@ -330,6 +358,11 @@ bool TryUnbundle(out IReadOnlyList<string>? replacement)
                 }
             }
 
+            bool PreviousTokenIsAnOptionExpectingAnArgument() =>
+                tokenList.Count > 1 &&
+                tokenList[tokenList.Count - 1] is { Type: TokenType.Option } optToken &&
+                currentCommand!.Children.GetByAlias(optToken.Value) is Option { Arity: { MaximumNumberOfValues: > 0 } };
```

Comment:
Are you sure that `currentCommand` cannot be null here? From looking at the two invocations I am not sure that is always guaranteed. Perhaps consider putting the null propagation back on it, as returning false in the null case appears to be correct anyway.

Reply:
Good call.